### PR TITLE
[Slack] Add anomalous event details

### PR DESCRIPTION
**Pull request by brachera**
Enhancement

## What does this PR do?

Adds steps to ingest pipeline to parse detail fields on events where action is anomaly

https://api.slack.com/admins/audit-logs-anomaly

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] `packages/slack/data_stream/audit/elasticsearch/ingest_pipeline/default.yml`

## How to test this PR locally

Sample log provided in: `packages/slack/data_stream/audit/_dev/test/pipeline/test-audit.log`

### CODE REVIEW

**Comment by efd6** [comment]
The username here verifies as having signed the CLA, but the PR itself does not reflect that. I suspect that the email address in the commit does not match the email address that you used to sign the CLA.

File: packages/slack/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.2.2"
+  changes:
+    - description: Adds anomaly detail fields
+      type: enhancement
+      link:
```
Comment (P1llus) [inline packages/slack/changelog.yml:6]: Could you add a link to this PR here?

You also need to bump the version in the top manifest.yml from 0.2.1 to 0.2.2

**Review by P1llus — COMMENTED** [review]
PR looks great! Added a small comment, and after that we should be able to merge.

**Comment by brachera** [comment]
Thanks
I'll add the PR and bump the version shortly
Got a change I need to make this evening around handling the nanoseconds from Slack API

**Comment by brachera** [comment]
👍

**Comment by jamiehynds** [comment]
@efd6 @P1llus are there any blockers to merging this PR?

File: packages/slack/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -120,6 +120,36 @@ processors:
     field: json.context.ip_address
     target_field: source.address
     ignore_missing: true
+- rename:
+    field: json.details.location
+    target_field: slack.audit.details.location
+    if: ctx?.event?.action == "anomaly"
```
Comment (efd6) [inline packages/slack/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:126]: `s/ctx\?/ctx/g`

**Review by efd6 — COMMENTED** [review]
Needs conflicts resolved.

File: packages/slack/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -120,6 +120,36 @@ processors:
...
+    field: json.details.previous_ua
+    target_field: slack.audit.details.previous_user_agent
+    if: ctx?.event?.action == "anomaly"
+    ignore_missing: true
+- rename:
+    field: json.details.reason
+    target_field: slack.audit.details.reason
+    if: ctx?.event?.action == "anomaly"
+    ignore_missing: true
+# Converts Slack microsecond output
```
Comment (andrewkroh) [inline packages/slack/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:143]: The sample value given in test-audit.log is `1673631621862`. That appears to be epoch milliseconds (2023-05-31T13:21:44). So we can remove this conversion.

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**